src/annotator/anchoring/test/range-test.js

```diff
@@ -0,0 +1,532 @@
+            result: 'ex',
+          },
+          {
+            startOffset: 0,
+            endOffset: 7,
+            result: 'text 1',
+          },
+        ].forEach(test => {
+          it('crops the text node appropriately', () => {
+            const browserRange = createBrowserRange({
+              startContainer: container.querySelector('#p-1').firstChild,
+              endContainer: container.querySelector('#p-1').firstChild,
+              startOffset: test.startOffset,
+              endOffset: test.endOffset,
+            });
+            const normalizedRange = browserRange.normalize();
+            assert.equal(normalizedRange.start.data, test.result);
+          });
+        });
+      });
+
+      context('common ancestor is not ELEMENT_NODE', () => {
+        it('corrects the commons ancestor to the first non text node', () => {
+          const browserRange = createBrowserRange({
+            commonAncestorContainer: container.querySelector('#p-1').firstChild,
+          });
+          const normalizedRange = browserRange.normalize();
+          assert.deepEqual(
+            normalizedRange.commonAncestor,
+            container.querySelector('#p-1')
+          );
+        });
+      });
+    });
+
+    describe('#serialize', () => {
+      it('converts BrowserRange to SerializedRange instance', () => {
+        const browserRange = createBrowserRange();
+        const result = browserRange.serialize(container);
+        assert.isTrue(result instanceof Range.SerializedRange);
+        assert.deepEqual(result, {
+          start: '/section[1]/p[1]',
+          startOffset: 0,
+          end: '/section[1]/span[1]/p[1]',
+          endOffset: 1,
+        });
+      });
+
+      it('converts BrowserRange to SerializedRange instance with `ignoreSelector` condition', () => {
+        const browserRange = createBrowserRange();
+        const result = browserRange.serialize(container, '#p-3');
+        // End xpath shall ignore the provided selector.
+        assert.deepEqual(result, {
+          start: '/section[1]/p[1]',
+          startOffset: 0,
+          end: '/section[1]/span[1]',
+          endOffset: 1,
+        });
+      });
+    });
+  });
+
+  describe('NormalizedRange', () => {
+    describe('#limit, #text and #textNodes methods', () => {
+      it('does not limit range if the limit node resides outside of bounds', () => {
+        const limit = createNormalizedRange().limit(
+          container.querySelector('#span-2')
+        );
+        assert.isNull(limit);
+      });
+
+      [
+        {
+          bound: '#p-1',
+          textNodes: ['text 1'],
+        },
+        {
+          bound: '#p-2',
+          textNodes: ['text 2a', 'text 2b'],
+        },
+        {
+          bound: '#section-1',
+          textNodes: ['text 1', 'text 2a', 'text 2b', 'text 3'],
+        },
+      ].forEach(test => {
+        it('limits range to the bounding node', () => {
+          let limit = createNormalizedRange().limit(
```

Comment:
```suggestion
          const limit = createNormalizedRange().limit(
```

Doesn't look like this is re-assigned.